ddtrace/contrib/internal/openai/_endpoint_hooks.py

```diff
@@ -715,3 +716,83 @@ def _record_response(self, pin, integration, span, args, kwargs, resp, error):
         else:
             span.set_metric("openai.response.total_bytes", getattr(resp, "total_bytes", 0))
         return resp
+
+
+class _ResponseHook(_EndpointHook):
+    _request_arg_params = ("api_key", "api_base", "api_type", "request_id", "api_version", "organization")
+    _request_kwarg_params = (
+        "model",
+        "include",
+        "instructions",
+        "max_output_tokens",
+        "metadata",
+        "parallel_tool_calls",
+        "previous_response_id",
+        "reasoning",
+        "service_tier",
+        "store",
+        "stream",
+        "temperature",
+        "text",
+        "tool_choice",
+        "tools",
+        "top_p",
+        "truncation",
+        "user",
+    )
+    _response_attrs = ("created_at", "id", "model", "tools")
+    ENDPOINT_NAME = "responses"
+    HTTP_METHOD_TYPE = "POST"
+    OPERATION_ID = "createResponseCompletion"
+
+    def _record_request(self, pin, integration, instance, span, args, kwargs):
+        super()._record_request(pin, integration, instance, span, args, kwargs)
+
+        input_data = kwargs.get("input", [])
+        if input_data:
+            if isinstance(input_data, str):
+                input_data = [input_data]
+
+            span._set_ctx_item("llmobs.response.input", input_data)
+
+    def _record_response(self, pin, integration, span, args, kwargs, resp, error):
+        resp = super()._record_response(pin, integration, span, args, kwargs, resp, error)
+        span._set_ctx_item(SPAN_KIND, "llm")
+
+        if not resp:
+            return resp
+
+        def handle_response_tools(resp, span):
+            if getattr(resp, "tools", None):
```

Comment:
If we're not doing anything if `resp.tools` is None, then let's just return early and avoid nested indents.